src/studio/ui/LineChart.java

```diff
@@ -42,144 +60,109 @@ public LineChart(KTableModel table) {
-                                        if (o instanceof ToDouble)
-                                            series.addOrUpdate(day,((ToDouble) o).toDouble());
+                                Day day = new Day(date.toDate(), tz);
+                                addOrUpdate(table, series, row, col, day);
                             }
-                        }
-                        else if (klass == K.KTimeVector.class) {
-                            series = new TimeSeries(table.getColumnName(col),Millisecond.class);
+                        } else if (klass == K.KTimeVector.class) {
+                            series = new TimeSeries(table.getColumnName(col), Millisecond.class);
 
                             K.KTimeVector times = (K.KTimeVector) table.getColumn(0);
-                            for (int row = 0;row < table.getRowCount();row++) {
+                            for (int row = 0; row < table.getRowCount(); row++) {
                                 K.KTime time = (K.KTime) times.at(row);
-                                Millisecond ms = new Millisecond(time.toTime(),tz);
-
-                                Object o = table.getValueAt(row,col);
-                                if (o instanceof K.KBase)
-                                    if (!((K.KBase) o).isNull())
-                                        if (o instanceof ToDouble)
-                                            series.addOrUpdate(ms,((ToDouble) o).toDouble());
+                                Millisecond ms = new Millisecond(time.toTime(), tz);
+                                addOrUpdate(table, series, row, col, ms);
                             }
-                        }
-                        else if (klass == K.KTimestampVector.class) {
-                            series = new TimeSeries(table.getColumnName(col),Day.class);
+                        } else if (klass == K.KTimestampVector.class) {
+                            series = new TimeSeries(table.getColumnName(col), Millisecond.class);
                             K.KTimestampVector dates = (K.KTimestampVector) table.getColumn(0);
 
-                            for (int row = 0;row < dates.getLength();row++) {
+                            for (int row = 0; row < dates.getLength(); row++) {
                                 K.KTimestamp date = (K.KTimestamp) dates.at(row);
-                                Day day = new Day(new java.util.Date(date.toTimestamp().getTime()),tz);
-
-                                Object o = table.getValueAt(row,col);
-                                if (o instanceof K.KBase)
-                                    if (!((K.KBase) o).isNull())
-                                        if (o instanceof ToDouble)
-                                            series.addOrUpdate(day,((ToDouble) o).toDouble());
+                                Millisecond day = new Millisecond(new java.util.Date(date.toTimestamp().getTime()), tz);

```

Comment:
That's the Day -> Millisecond change - the rest is just extracting common code in addOrUpdate